src/internal/list.js

```diff
@@ -0,0 +1,2 @@
+export const empty = ({isEmpty: true, size: 0, head: null, tail: null});
+export const cons = (head, tail) => ({isEmpty: false, size: tail.size + 1, head, tail});
```

Comment:
Beautiful! :unicorn:

The last thing to consider is `s/empty/nil/` to avoid confusion with `Z.empty`.

The parens around the object literal assigned to `empty` are not required (although you may have included them for aesthetic reasons).